### DISCUSSION TITLE
Can we detect when our app is uninstalled from within Amazon?

### DISCUSSION THREAD

**Discussion by kaixilee** [amzn/selling-partner-api-models#3768] [General]
Hello, is there a way to know if a user has uninstalled our app via the AmazonSelling Partner Appstore side (so that we can delete that particular installation for our integration)?

**Comment by stefnats** [comment]
Only thing you could check is if the credentials have been revoked by doing an API call. if it returns an invalid_grant error, the permissions have been revoked.
I am not sure how it is related to uninstalling an app store application, but this would be my approach.
